### DISCUSSION TITLE
Broken wiki, no place to download?

### DISCUSSION THREAD

**Discussion by colingoodman** [inducer/pycuda#374] [Troubleshooting] [answered]
Pip3 failed to install pycuda, so I found the linked wiki to troubleshoot. [It explains](https://wiki.tiker.net/PyCuda/Installation/Linux/) I need to download a tgz pycuda file, but all of the links appear to be broken:

![Screen Shot 2022-08-22 at 10 44 36 AM](https://user-images.githubusercontent.com/29467620/185975024-0d4d80d1-f1f1-4cce-a06b-a76ce33b899c.png)

This is the case on both the linux and mac pages of the wiki.

**Comment by inducer** [comment, marked as answer]
I agree that the Wiki situation is a bit sad. What I did there was a minimal job to salvage some wiki pages from a moinmoin wiki that was on the way out. That documentation should be moved to the docs.

As for download, just like pip, download from the package index:

https://pypi.org/project/pycuda/

And if you run into installation trouble, feel free to post here (with detailed error messages).

**Reply by colingoodman** [reply]
Thanks for the help. When attempting to install via pip, I get `ERROR: Could not build wheels for pycuda, which is required to install pyproject.toml-based projects`

I've installed wheel via pip as well as the following apt packages: `software-properties-common libboost-all-dev python-numpy python-setuptools libboost-python-dev libboost-thread-dev python3-dev`

I am attempting to install pycuda into a docker container via dockerfile.

**Reply by colingoodman** [reply]
Attempting to install pycuda onto a Nvidia jetson device on jetpack 5, I receive some more verbose feedback from pip.

Some example stuff:

```      In file included from src/cpp/cuda.cpp:4:
      src/cpp/cuda.hpp:14:10: fatal error: cuda.h: No such file or directory
         14 | #include <cuda.h>
            |          ^~~~~~~~
      compilation terminated.
      /tmp/pip-build-env-hn6xrr2x/overlay/lib/python3.8/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'pycuda.cuda' as data is deprecated, please list it in `packages`.
```

```
Python recognizes 'pycuda.cuda' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
```

```
ERROR: Could not build wheels for pycuda, which is required to install pyproject.toml-based projects
```

**Reply by mitkotak** [reply]
Can you try ?
```
pip install git+https://github.com/inducer/pycuda.git@main
```

**Reply by colingoodman** [reply]
@mitkotak Thanks for the suggestion, unfortunately it did not change anything. 

I did find [a forum thread](https://forums.developer.nvidia.com/t/problem-installing-pycuda-with-agx-orin/220109/4) that suggests setting some environment variables. This fixed the installation on my Nvidia device, but not on my Docker build. This implies to me that there is a lack of environment parity with the docker build and I thus need to configure/install something else.
